src/screens/ProfileScreen/elements/TextWithEmoji.tsx

```diff
@@ -13,9 +16,19 @@ export function TextWithEmoji({
   textStyle: object;
   testId: string;
 }) {
+  const sanitizeNewLine = text?.split('\n');
+
   return (
     <Text style={[styles.text, textStyle]} testID={testId}>
-      {text?.split(' ').map((item: string, index: number) => {
+      {sanitizeNewLine?.map((item: string, index: number) => {
+        if (index < sanitizeNewLine?.length - 1) {
+          item += '\n';
+        }
+
+        if (isValidUrl(item)) {
+          return <LinkableText text={item} key={`${item}-${index}`} />;
+        }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in this section has been modified to split the text by new lines and check for valid URLs. If a valid URL is found, it is rendered as a `LinkableText` component. This is a good change as it improves the handling of URLs within the text.
```diff
-   return (
-     <Text style={[styles.text, textStyle]} testID={testId}>
-      {text?.split(' ').map((item: string, index: number) => {
-         return (
-           <Text
-             key={index}
+   const sanitizeNewLine = text?.split('\n');
+
+   return (
+     <Text style={[styles.text, textStyle]} testID={testId}>
+       {sanitizeNewLine?.map((item: string, index: number) => {
+         if (index < sanitizeNewLine?.length - 1) {
+           item += '\n';
+         }
+
+         if (isValidUrl(item)) {
+           return <LinkableText text={item} key={`${item}-${index}`} />;
+         }
+
+         return (
+           <Text
+             key={index}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->